src/darwin/Framework/CHIP/CHIPP256KeypairBridge.mm

```diff
@@ -0,0 +1,143 @@
+    }
+
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::ECDSA_sign_msg(const uint8_t * msg, size_t msg_length, P256ECDSASignature & out_signature)
+{
+    if (!HasKeypair()) {
+        CHIP_LOG_ERROR("ECDSA sign msg failure: no keypair to sign with.");
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+    NSData * msgData = [NSData dataWithBytes:msg length:msg_length];
+    NSMutableData * hashedData = [NSMutableData dataWithLength:CC_SHA256_DIGEST_LENGTH];
+    CC_SHA256(msgData.bytes, (CC_LONG) msgData.length, (unsigned char *) hashedData.mutableBytes);
+    CHIP_LOG_DEBUG("Generated Msg hash, signing hash now");
+    NSData * signature = [mKeypair ECDSA_sign_hash:hashedData];
+    if (!signature) {
+        CHIP_LOG_ERROR("ECDSA sign msg failure: no signature returned");
+        return CHIP_ERROR_INTERNAL;
+    }
+    if (signature.length > out_signature.Capacity()) {
+        CHIP_LOG_ERROR("ECDSA sign msg failure: unexpected signature size %tu vs %tu ", signature.length, out_signature.Capacity());
+        return CHIP_ERROR_NO_MEMORY;
+    }
+    std::memmove(out_signature, signature.bytes, signature.length);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::ECDSA_sign_hash(const uint8_t * hash, size_t hash_length, P256ECDSASignature & out_signature)
+{
+    if (!HasKeypair()) {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+    NSData * hashData = [NSData dataWithBytes:hash length:hash_length];
+    NSData * signature = [mKeypair ECDSA_sign_hash:hashData];
+    if (!signature) {
+        CHIP_LOG_ERROR("ECDSA sign hash failure: no signature returned");
+        return CHIP_ERROR_INTERNAL;
+    }
+    if (signature.length > out_signature.Capacity()) {
+        CHIP_LOG_ERROR(
+            "ECDSA sign hash failure: unexpected signature size %tu vs %tu ", signature.length, out_signature.Capacity());
+        return CHIP_ERROR_NO_MEMORY;
+    }
+    std::memmove(out_signature, signature.bytes, signature.length);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::ECDH_derive_secret(
+    const P256PublicKey & remote_public_key, P256ECDHDerivedSecret & out_secret) const
+{
+    if (!HasKeypair()) {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::setPubkey()
+{
+    SecKeyRef pubkeyRef = [mKeypair pubkey];
+    if (!pubkeyRef) {
+        CHIP_LOG_ERROR("Unable to initialize Pubkey");
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    NSData * pubkeyData = (__bridge_transfer NSData *) SecKeyCopyExternalRepresentation(pubkeyRef, nil);
+    if (!pubkeyData) {
+        CHIP_LOG_ERROR("Unable to copy external representation for pubkey ref, returning empty Pubkey");
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    uint8_t pubkeyBytes[kP256_PublicKey_Length];
+    std::memmove(&pubkeyBytes, pubkeyData.bytes, pubkeyData.length);
+    mPubkey = P256PublicKey(pubkeyBytes);
```

Comment:
Why can't use use pubkeyData.bytes directly here? We should be able to create a byte span to use here and not require a temporary copy just to auto-template the size. Maybe `FixedSpan<kP256_PublicKey_Length>(pubkeyData.bytes)` works or something similar.